### ISSUE TITLE
Is there a replacement for the U3?

### ISSUE THREAD

**Issue by dippaolo** [SukkoPera/OpenFlops#13]
Is there a replacement for the U3?

Access to the STM32F105RBT6 microcontroller is difficult.

You can get it, but it's expensive :(

**Comment by SukkoPera** [comment]
No, feel free to mod the design to use a different IC.

**Comment by Bytedecode72** [comment]
i've got some GD32F105RBT6 coming instead of the STM32F105RBT6

**Comment by SukkoPera** [comment]
If it works, can you please provide a link to where those can be bought?

**Comment by ra1nb0w** [comment]
@Bytedecode72 any news with GD32F105RBT6? thank you

**Comment by ra1nb0w** [comment]
seems that also `AT32F415RCT7` (from new GOTEK boards) is usable but not findable in the market

**Comment by Bytedecode72** [comment]
> @Bytedecode72 any news with GD32F105RBT6? thank you

i got the  chips just need to order other components and assemble then get debug firmware on there to see if anything is tripping it up.

**Comment by ra1nb0w** [comment]
Thank you @Bytedecode72 for the update. Report back when it will work (or not).

**Comment by Bytedecode72** [comment]
> Thank you @Bytedecode72 for the update. Report back when it will work (or not).

i will do just waiting on the components now all sourc
![bom](https://user-images.githubusercontent.com/83699521/160129894-4bd959b5-20d9-4484-bfc4-01934e42a4cd.png)
e

**Comment by ra1nb0w** [comment]
I bought an GD32F105RBT6 too. We will see ;-)

**Comment by SzymonGk** [comment]
> > Thank you @Bytedecode72 for the update. Report back when it will work (or not).
> 
> i will do just waiting on the components now all sourc ![bom](https://user-images.githubusercontent.com/83699521/160129894-4bd959b5-20d9-4484-bfc4-01934e42a4cd.png) e

Hi, any update regarding GD32F105RBT6?

**Comment by SzymonGk** [comment]
I do not think that GD32F105RBT6 is compatible substitution for STM32F105RBT6. I soldered IC to the board and programmed with hex flie provided for STM32F105RBT6 (FlashFloppy), unfortunately device does not work. I programmed it via two methods, first via Demonstrator Gui and second via StLink2 soldering two thin wires to pins 46(SWDIO) and 49(SWCLK). I tried also Giga Devices programming software but it crashes when hex file selected.

**Comment by SukkoPera** [comment]
It should be compatible but it requires a dedicated firmware, I think. Please see the FlashFloppy project page.

**Comment by jduraes** [comment]
> > Thank you [@Bytedecode72](https://github.com/Bytedecode72) for the update. Report back when it will work (or not).
> 
> i will do just waiting on the components now all sourc ![bom](https://user-images.githubusercontent.com/83699521/160129894-4bd959b5-20d9-4484-bfc4-01934e42a4cd.png) e

Would you be so kind to share your BOM file, unless one already exists somewhere? Saves typing it and avoiding mistakes. Thanks.

**Comment by Bytedecode72** [comment]
> > > Thank you [@Bytedecode72](https://github.com/Bytedecode72) for the update. Report back when it will work (or not).
> > 
> > 
> > i will do just waiting on the components now all sourc ![bom](https://user-images.githubusercontent.com/83699521/160129894-4bd959b5-20d9-4484-bfc4-01934e42a4cd.png) e
> 
> Would you be so kind to share your BOM file, unless one already exists somewhere? Saves typing it and avoiding mistakes. Thanks.

[openflops_1-2_11 boardsbom.zip](https://github.com/user-attachments/files/18492082/openflops_1-2_11.boardsbom.zip)

i haven`t been able to do any assembly due to health issues and numb fingers, but i`ll attached the excel bom file

**Comment by jduraes** [comment]
> > 
> > Would you be so kind to share your BOM file, unless one already exists somewhere? Saves typing it and avoiding mistakes. Thanks.
> 
> [openflops_1-2_11 boardsbom.zip](https://github.com/user-attachments/files/18492082/openflops_1-2_11.boardsbom.zip)
> 
> i haven`t been able to do any assembly due to health issues and numb fingers, but i`ve attached the excel bom file, if you change number of boards to build it will update the required amount of components.

Thank you very much. 

I know the numb hands feeling all too well myself, as I can only do about an hour or so of soldering. Hope you get well on that.

**Comment by gilmod714** [comment]
> If it works, can you please provide a link to where those can be bought?

I found some cheap GD32F10 on Aliexpress a while back ..  not sure if that'll work .. I was reading the previous issues , did they work? or they need special firmware?
